src/svg.rs

```diff
@@ -33,13 +33,15 @@ impl SvgVersion {
 declare_surface!(SvgSurface, SurfaceType::Svg);
 
 impl SvgSurface {
-    pub fn new<P: AsRef<Path>>(width: f64, height: f64, path: P) -> Result<SvgSurface, Status> {
-        let path = path.as_ref().to_string_lossy().into_owned();
-        let path = CString::new(path).unwrap();
-
+    pub fn new<P: AsRef<Path>>(
+        width: f64,
+        height: f64,
+        path: Option<P>,
+    ) -> Result<SvgSurface, Status> {
+        let path = path.map(|p| p.as_ref().to_string_lossy().into_owned());
```

Comment:
```suggestion
        let path = path.map(|p| CString::new(path.as_ref().as_os_str().as_bytes()));
        let path = path.map(|p| p.as_ptr()).unwrap_or(ptr::null());
```

Ideally you'd do more than that though (Windows-specific special case...): https://github.com/gtk-rs/glib/blob/faff6573602daec764584030e959f7d28a0aef81/src/translate.rs#L421-L457